php/ec/.storybook/webpack.config.js

```diff
@@ -1,15 +1,12 @@
-module.exports = ({ config, mode }) => {
+module.exports = ({ config: defaultConfig, mode }) => {
   // Make it less verbose
   if (mode === 'PRODUCTION') {
-    // Remove ProgressPlugin (4th plugin)
-    const plugin = config.plugins.splice(3, 1);
-    if (!(plugin[0].constructor.name === 'ProgressPlugin')) {
-      console.error(
-        'Error: 4th plugin is not ProgressPlugin.\nCheck src/systems/ec/implementations/react/storybook/.storybook/webpack.config.js'
-      );
-      return process.exit(1);
+    // Remove ProgressPlugin (5th plugin)
+    const plugin = defaultConfig.plugins.splice(4, 1);
+    if (plugin[0].constructor.name !== 'ProgressPlugin') {
+      defaultConfig.plugins.push(plugin[0]);
```

Comment:
Yeah, for a long time i didn't even know exactly what was this supposed to do, but when you see the result without it it becomes very clear, for now i just "fixed" it so that it works but i agree with you, a better approach could be found, although i remember there was an attempt in the past about looping in the plugins to find Process instead of assuming it was in a certain position of the array, but it was not working and we reverted back to this.
We can give it another try, maybe in a separated ticket ;)